app/lib/actions/katello/host/hypervisors_update.rb

```diff
@@ -142,10 +142,16 @@ def duplicate_name(hypervisor, consumer)
 
         def create_host_for_hypervisor(name, organization, location = nil)
           location ||= Location.default_host_subscribe_location!
-          host = ::Host::Managed.new(:name => name, :organization => organization,
-                                     :location => location, :managed => false, :enabled => false)
-          host.save!
-          host
+          begin
+            host = ::Host::Managed.new(:name => name, :organization => organization,
+                                       :location => location, :managed => false, :enabled => false)
+            host.save!
+          rescue
+            # Taken from Foreman's active record error message
+            raise ::Katello::Errors::InvalidVirtWhoHost, "The hostname #{name} can contain only lowercase letters, numbers, dashes and dots according to RFC921, RFC952 and RFC1123"
```

Comment:
This should probably be a regular Rails validation, not a rescue.

Check out `app/models/katello/concerns/organization_extensions.rb`, see how it has an included block and then has some validations added to the Organization model?

```rb
included do
...
validates_with Validators::KatelloLabelFormatValidator, :attributes => :label
validates :label, :uniqueness => true
```

In this case we could do the same thing, except add it to `host_managed_extensions`. That would allow us to add validations to `Host::Managed`.

As for the validation itself, you can look at

https://github.com/Katello/katello/blob/master/app/models/katello/root_repository.rb#L88

and

https://github.com/Katello/katello/blob/master/app/models/katello/root_repository.rb#L182C5-L187C8

and write something similar to that.